File: 67/Solution.java
class Solution {
    public String addBinary(String a, String b) {
        int i = a.length() - 1;
        int j = b.length() - 1;
        StringBuilder res = new StringBuilder();
        int added = 0;
        while (i >= 0 || j >= 0) {
            if (i >= 0) {
                added += a.charAt(i);
                i--;
            }
            if (j >= 0) {
                added += b.charAt(j) - '0';
                j--;
            }
            res.append(Integer.toString(added % 2));
            added /= 2;
        }
        if (added == 1) {
            res.append('1');
        }
        return res.reverse().toString();
    }
}

File: 67/solution.py
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        i = len(a) - 1
        j = len(b) - 1
        res = ""
        added = 0
        while i >= 0 or j >= 0:
            if i >= 0:
                added += int(a[i])
                i -= 1
            if j >= 0:
                added += int(b[j])
                j -= 1
            res = str(added % 2) + res
            added //= 2
        if added:
            return "1" + res
        return res
